src/mam/mod_mam_riak_timed_arch_yz.erl

```diff
@@ -122,20 +122,26 @@ stop_muc_archive(Host) ->
     ejabberd_hooks:delete(mam_muc_remove_archive, Host, ?MODULE, remove_archive, 50),
     ok.
 
+%% LocJID - From/To JID depending on direction of message (single message is saved in two directions)
+%% RemJID - To/From JID depending on direction of message (single message is saved in two directions)
+%% SrcJID - "Real" sender JID
 archive_message(_Result, Host, MessId, _UserID, LocJID, RemJID, SrcJID, _Dir, Packet) ->
     try
-        archive_message(Host, MessId, LocJID, RemJID, SrcJID, Packet, pm)
+        archive_message(Host, MessId, LocJID, RemJID, SrcJID, LocJID, Packet, pm)
     catch _Type:Reason ->
             ?WARNING_MSG("Could not write message to archive, reason: ~p",
                          [{Reason, erlang:get_stacktrace()}]),
             ejabberd_hooks:run(mam_drop_message, Host, [Host]),
             {error, Reason}
     end.
 
-archive_message_muc(_Result, Host, MessId, _UserID, LocJID, _FromJID, SrcJID, _Dir, Packet) ->
+%% LocJID - MUC/MUC Light room's JID
+%% FromJID - "Real" sender JID
+%% SrcJID - Full JID of user within room (room@domain/user)
```

Comment:
`room@domain/nickname`, because `user` may be confused with username

Reply:
I actually didn't want to use `nickname` as this may also be JID (I think) in case of MUC Light. I agree with the fact that`user` is not the best choice, but `nickname` doesn't feel generic enough.

Reply:
OK, fair enough. :)